include/deal.II/fe/fe.h

```diff
@@ -654,9 +654,17 @@ class FiniteElement : public Subscriptor,
    * ExcShapeFunctionNotPrimitive. In that case, use the
    * shape_value_component() function.
    *
-   * An ExcUnitShapeValuesDoNotExist is thrown if the shape values of the
-   * FiniteElement under consideration depends on the shape of the cell in
-   * real space.
+   * Implementations of this function should throw an exception of type
+   * ExcUnitShapeValuesDoNotExist if the shape functions of the
+   * FiniteElement under consideration depend on the shape of the cell in
+   * real space, i.e., if the shape functions are not defined by
+   * mapping from the reference cell. Some non-conforming elements are
+   * defined this way, as is the FE_DGPNonparametric class, to name just
+   * one example.
```

Comment:
That's actually a bug in that class then. It doesn't make sense for this class to export shape function values on the reference cell because shape functions are not defined there. I think the text here will be correct once that is fixed.

I'll open a bug report that describes that problem and will take care of it in the FE_DGPNonparametric class.